### PR TITLE
Feature/locker 1158

### PR DESCRIPTION
**Pull request by muralikuppoor**
Create Admin GUI - just for admins

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ARG home=/srv
+
+ENV PING_TTL=$PING_TTL
+ENV LOGSTASH_HOST=$LOGSTASH_HOST
+ENV SERVER_PORT=$SERVER_PORT
+ENV AIRSIDE_DEBUG=$AIRSIDE_DEBUG
+ENV SPRING_PROFILES_ACTIVE=$SPRING_PROFILES_ACTIVE
+ENV API_USER=$API_USER
+ENV API_PASSWORD=$API_PASSWORD
+ENV COMMERCE_API_URL=$COMMERCE_API_URL
```
Comment (syedwali) [inline Dockerfile]: COMMERCE?

**Comment by syedwali** [comment]
Yikes! That is a lot of lines of code. 

Let's plan to sit down and touch on a few things here that we can keep out of the 'master' commit. The promotions code / commerce service references, etc.

File: Dockerfile (hunk 1 above)
Comment (muralikuppoor) [reply Dockerfile]: Removed Commerce module

File: airside-security-console-backend/src/main/java/net/airside/security/configuration/AirsideOAuthFilter.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  private String redirectUri;
+
+  @Value("${proof.application.accessTokenUri}")
+  private String accessTokenUri;
+
+  @Value("${proof.application.userAuthorizationUri}")
+  private String userAuthorizationUri;
+
+  @Value("${proof.application.codeVerifier}")
+  private String codeVerifier;
```
Comment (rajairside) [inline airside-security-console-backend/src/main/java/net/airside/security/configuration/AirsideOAuthFilter.java]: code verifier is not a static value.  It's a dynamic value.

File: airside-security-console-backend/src/main/resources/logback-spring.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          <timestamp/>
+          <mdc/>
+          <context/>
+          <version/>
+          <logLevel/>
+          <loggerName/>
+          <pattern>
+            <pattern>
+              {
+              "appName": "commerce-web",
```
Comment (rajairside) [inline airside-security-console-backend/src/main/resources/logback-spring.xml]: rename this please.

File: airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  @Test
+  public void testGetUserInfo() {
+    assertNotNull(userInfoController.getUserInfo().getEmail());
+  }
+
+  @Test
+  public void testGetUserInfoWithUserInfoObject() {
+    UserInfoTO userInfoTO = new UserInfoTO();
+    userInfoTO.setEmail("albert.micheal@airsidemobile.com");
+    assertNotNull(userInfoController.getUserInfoTO(userInfoTO).getEmail());
```
Comment (rajairside) [inline airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java]: I don't understand this test.  Are we testing UserInfoTO or userInfoController?

File: airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    UserInfoTO userInfoTO = new UserInfoTO();
+    userInfoTO.setEmail("test@airsidemobile.com");
+    assertNotNull(userInfoController.getUserInfoTO(userInfoTO).getEmail());
+  }
+
+  @Test
+  public void testGetUserInfoWithUserInfoObjectNoEmail() {
+    UserInfoTO userInfoTO = new UserInfoTO();
+    userInfoTO.setEmail("murali.kuppoor@airsidemobile.com");
+    assertNotNull(userInfoController.getUserInfoTO(userInfoTO).getEmail());
```
Comment (rajairside) [inline airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java]: 3 similar tests - what are we testing here?

File: airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+    assertNotNull(ApplicationConstants.BASIC);
+    assertNotNull(ApplicationConstants.COLON);
+    assertNotNull(ApplicationConstants.ID_TOKEN);
+    assertNotNull(ApplicationConstants.KID);
+    assertNotNull(ApplicationConstants.LOCAL);
+    assertNotNull(ApplicationConstants.PARAMETERS);
+    assertNotNull(ApplicationConstants.READ_ONLY);
+    assertNotNull(ApplicationConstants.US_ASCII);
+    assertNotNull(ApplicationConstants.X_REQUEST_ID);
+    assertNotNull(ApplicationConstants.DELETED);
```
Comment (rajairside) [inline airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java]: what is under test here?

File: airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+    assertTrue(summaryRequestTO.getRowLimit() == 10);
+    assertTrue(summaryRequestTO.getSearchString().equals("sony"));
+    assertTrue(summaryRequestTO.getExpiryInDays() == 1);
+    assertTrue(!summaryRequestTO.isOverSubscribed());
+    assertTrue(!summaryRequestTO.isMaximumUsed());
+  }
+
+  @Test
+  public void testLogout() {
+    assertNotNull(userInfoController.logout(httpServletRequest));
```
Comment (rajairside) [inline airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java]: Looks like these tests are written to just get code coverage but they do not reflect true reality.
In such cases - selenium tests should be used to actually login and have the user logout.

File: airside-security-console-ui/src/app/models/ProductDescriptor.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+export interface ProductDescriptor {
+  storeSubscriptionListingId: string;
```
Comment (rajairside) [inline airside-security-console-ui/src/app/models/ProductDescriptor.ts]: you can remove this ?

File: airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+  }
+
+  @Test
+  public void testApplicationProperties() {
+
+    applicationProperties.setAdmins(applicationProperties.getAdmins());
```
Comment (rajairside) [inline airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java]: ??

File: airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+    applicationProperties.setLogoutURL(applicationProperties.getLogoutURL());
+
+    assertNotNull(applicationProperties.getAdmins());
```
Comment (rajairside) [inline airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java]: This is not a valid test case.

File: Dockerfile (hunk 1 above)
Comment (syedwali) [reply Dockerfile]: Awesome. This gets in line with the Security Team consensus

File: airside-security-console-backend/src/main/java/net/airside/security/configuration/OAuthSecurityConfig.java [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.security.web.authentication.LoginUrlAuthenticationEntryPoint;
+import org.springframework.security.web.authentication.preauth.AbstractPreAuthenticatedProcessingFilter;
+import org.springframework.security.web.csrf.CsrfTokenRepository;
+import org.springframework.security.web.csrf.HttpSessionCsrfTokenRepository;
+
+@Configuration
+@EnableWebSecurity
+@Profile({"airsideOAuth"})
+@Order(2)
+public class OAuthSecurityConfig extends WebSecurityConfigurerAdapter {
```
[the remark this answers is not in the document]
Comment (muralikuppoor) [reply airside-security-console-backend/src/main/java/net/airside/security/configuration/OAuthSecurityConfig.java]: I think we should move this commons after first snapshot release to Dev-environment

File: airside-security-console-backend/src/main/java/net/airside/security/configuration/AirsideOAuthFilter.java (hunk 2 above)
Comment (muralikuppoor) [reply airside-security-console-backend/src/main/java/net/airside/security/configuration/AirsideOAuthFilter.java]: Good Catch! Implemented

File: airside-security-console-backend/src/main/java/net/airside/security/configuration/AirsideOAuthFilter.java [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+
+  @Value("${proof.application.codeChallenge}")
+  private String codeChallenge;
```
[the remark this answers is not in the document]
Comment (muralikuppoor) [reply airside-security-console-backend/src/main/java/net/airside/security/configuration/AirsideOAuthFilter.java]: done

File: airside-security-console-backend/src/main/java/net/airside/security/configuration/OAuthSecurityConfig.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (muralikuppoor) [reply airside-security-console-backend/src/main/java/net/airside/security/configuration/OAuthSecurityConfig.java]: We'll see how it works on first deployment to dev and then we'll move to commons.

File: airside-security-console-backend/src/main/java/net/airside/security/controllers/CustomerInfoController.java [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+  @Autowired private ApplicationProperties applicationProperties;
+  @Autowired protected ObjectMapper objectMapper;
+
+  @GetMapping("login")
+  public @ResponseBody String greet(@RequestParam  String code) {
+    return "Welcome to Airside-Security-console "+ code;
+  }
+
+  @GetMapping("api/credlist1")
+  public @ResponseBody List<Credentials> getCredentialsList() throws IOException {
```
[the remark this answers is not in the document]
Comment (muralikuppoor) [reply airside-security-console-backend/src/main/java/net/airside/security/controllers/CustomerInfoController.java]: Will implement in next sprint

File: airside-security-console-backend/src/main/resources/logback-spring.xml (hunk 3 above)
Comment (muralikuppoor) [reply airside-security-console-backend/src/main/resources/logback-spring.xml]: Good catch! Done!

File: airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java (hunk 4 above)
Comment (muralikuppoor) [reply airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java]: Removed unwanted test cases as per your review comments!

File: airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java (hunk 7 above)
Comment (muralikuppoor) [reply airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java]: done

File: airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java (hunk 9 above)
Comment (muralikuppoor) [reply airside-security-console-backend/src/test/java/net/airside/security/controllers/UserInfoControllerTests.java]: Good catch! removed!

**Pull request merged**